### PR TITLE
Enable fragment result cache and data cache

### PR DESCRIPTION
**Pull request by agrawalreetika**
# Fragment result cache -
fragment cache -> disk = worker ram * 2 -> Effective cache size in config is disk * 0.95

# Alluxio Data cache -
data cache -> disk = worker ram * 3 -> Effective cache size in config is disk * 0.95

### CODE REVIEW

File: clusters/templates/workers/config.properties [hunk 1]
```diff
@@ -53,6 +53,13 @@ scale-writers=true
 shutdown.grace-period=30s
 sink.max-buffer-size=1GB
 
+# fragment result cache
```
Comment (ethanyzhang) [inline clusters/templates/workers/config.properties:56]: I talked to Reetika offline and she will make this section conditional

**Review by ethanyzhang — CHANGES_REQUESTED** [review]

**Review by ethanyzhang — CHANGES_REQUESTED** [review]

File: clusters/2xlarge/config.json [hunk 2]
```diff
@@ -3,8 +3,10 @@
     "coordinator_instance_type": "r6i.16xlarge",
     "coordinator_instance_ebs_size": 1024,
```
[the remark this answers is not in the document]
Comment (agrawalreetika) [reply clusters/2xlarge/config.json:4]: @wanglinsong Should we reduce this as well to 50GB? Or what do you suggest for this?

[the remark this answers is not in the document]
Comment (wanglinsong) [reply clusters/2xlarge/config.json:4]: 50G should be good enough for the OS.

File: clusters/templates/catalog/iceberg-trino.properties [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+hive.metastore-cache-ttl=240m
+hive.metastore-refresh-interval=30m
+hive.metastore-cache-maximum-size=1000000
+# writer properties
+iceberg.max-partitions-per-writer=5000
+
+# metadata caching
+hive.parquet.use-column-names=true
+hive.parquet-batch-read-optimization-enabled=true
+iceberg.parquet.metadata-cache-enabled=true
```
Comment (ethanyzhang) [inline clusters/templates/catalog/iceberg-trino.properties:18]: Actually I am not able to find this in Trino's source code..
https://github.com/search?q=repo%3Atrinodb%2Ftrino+metadata-cache-enabled&type=code

**Review by ethanyzhang — CHANGES_REQUESTED** [review]

**Review by ethanyzhang — APPROVED** [review]

**Pull request merged**